Touch Bar Simulator/TouchBarWindow.swift

```diff
@@ -45,6 +66,83 @@ final class TouchBarWindow: NSPanel {
 		}
 	}
 
+	var dockBehaviorTimer: Timer?
+
+	func startDockBehaviorTimer() {
+		stopDockBehaviorTimer()
+		// Throttles the function to only execute every 1/10 second.
+		// Without any throttling the function would be very expensive and use an average of 7% CPU (on my machine) which is too much IMO.
+		dockBehaviorTimer = Timer.scheduledTimer(timeInterval: 0.1, target: self, selector: #selector(handleDockBehavior), userInfo: nil, repeats: true)
+		dockBehaviorTimer!.fire()
+	}
+
+	func stopDockBehaviorTimer() {
+		guard dockBehaviorTimer != nil else {
+			return
+		}
+		dockBehaviorTimer!.invalidate()
+		dockBehaviorTimer = nil
+	}
+
+	var dockBehavior: Bool = defaults[.dockBehavior] {
+		didSet {
+			if dockBehavior {
+				guard self.docking != nil else {
+					return
+				}
+				guard self.docking! != .floating else {
```

Comment:
Would be better to be explicit and list the two modes you actually want instead of the inverse.